bugsnag-android-core/src/test/java/com/bugsnag/android/BugsnagJournalTest.kt

```diff
@@ -0,0 +1,148 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.internal.JournalKeys
+import org.junit.Assert
+import org.junit.ClassRule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+
+class BugsnagJournalTest {
+    companion object {
+        @ClassRule
+        @JvmField
+        val folder: TemporaryFolder = TemporaryFolder()
+    }
+    private val baseDocumentPath by lazy { File(folder.root, "mydocument") }
+
+    @Test
+    fun testNoOldDocument() {
+        val oldDocument = BugsnagJournal.loadPreviousDocument(File(baseDocumentPath, "doesNotExistDocument"))
+        Assert.assertNull(oldDocument)
+    }
+
+    @Test
+    fun testInitialDocumentContents() {
+        val observedDocument = BugsnagJournal.withInitialDocumentContents(mapOf())
+        val expectedDocument = mapOf<String, Any>(
+            JournalKeys.keyVersionInfo to mapOf(
+                JournalKeys.keyType to BugsnagJournal.journalType,
+                JournalKeys.keyVersion to BugsnagJournal.journalVersion
+            )
+        )
+        Assert.assertEquals(
+            BugsnagTestUtils.normalized(expectedDocument),
+            BugsnagTestUtils.normalized(observedDocument)
+        )
+    }
+
+    @Test
+    fun testEmptyDocument() {
+        val journal = BugsnagJournal(NoopLogger, baseDocumentPath, mapOf())
+        journal.snapshot()
+
+        val observedDocument = BugsnagJournal.loadPreviousDocument(baseDocumentPath)
+        val expectedDocument = BugsnagJournal.withInitialDocumentContents(mapOf())
+
+        Assert.assertEquals(
+            BugsnagTestUtils.normalized(expectedDocument),
+            BugsnagTestUtils.normalized(observedDocument)
+        )
+        Assert.assertEquals(
+            BugsnagTestUtils.normalized(expectedDocument),
+            BugsnagTestUtils.normalized(journal.document)
+        )
+    }
+
+    @Test
+    fun testJournalEntry() {
+        val journal = BugsnagJournal(NoopLogger, baseDocumentPath, mapOf())
+        journal.addCommand("x", 100)
+        journal.snapshot()
+
+        val observedDocument = BugsnagJournal.loadPreviousDocument(baseDocumentPath)
+        val expectedDocument = HashMap(BugsnagJournal.withInitialDocumentContents(mapOf()))
+        expectedDocument["x"] = 100
+
+        Assert.assertEquals(
+            BugsnagTestUtils.normalized(expectedDocument),
+            BugsnagTestUtils.normalized(observedDocument)
+        )
+        Assert.assertEquals(
+            BugsnagTestUtils.normalized(expectedDocument),
+            BugsnagTestUtils.normalized(journal.document)
+        )
+    }
+
+    @Test
+    fun testManyJournalEntries() {
+        val journal = BugsnagJournal(NoopLogger, baseDocumentPath, mapOf())
+        journal.addCommand("one hundred", 100)
+        journal.addCommand("three\\.four\\.five\\.slash\\\\slash.-1", "x")
```

Comment:
Maybe these could be raw strings?
```suggestion
        journal.addCommand("""three\.four\.five\.slash\\slash.-1""", "x")
```